src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1118,13 +1118,17 @@ internal static Assembly DefineTypes(Parser parser, Ast rootAst, TypeDefinitionA
                                       : rootAst.Extent.File
                                       .Replace('\\', (char)0x29f9)
                                       .Replace('/', (char)0x29f9)
-                                      .Replace(',', (char)0x201a)
                                       .Replace(':', (char)0x0589));
 
-            var assembly = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName(assemblyName),
+            // We can not use 'AssemblyName(string assemblyName)' constructor because it is trying to parse the assembly name
+            // as 'Name <,Culture = CultureInfo> <,Version = Major.Minor.Build.Revision> <, StrongName> <,PublicKeyToken> '\0''
+            // See notes in https://msdn.microsoft.com/en-us/library/system.reflection.assemblyname(v=vs.110).aspx
+            var assemblyObj = new AssemblyName();
+            assemblyObj.Name = assemblyName;
```

Comment:
This seems like a bypass that we shouldn't use. I'm worried that some random non-CLR code might parse the assembly name (e.g. PowerShell) in a manner similar to how the CLR does when clearly.

The goal here was to make it easy to map the assembly to the original source, but in some ways this already fails because of the path.

A better fix is probably to use a simple name like `PowerShell Dynamic Class` possibly with a counter, but to **add** a constructor to `DynamicClassImplementationAttribute` and pass the filename as an argument to that attribute.

This way we have a non-hacky way of mapping assemblies to script files.

Reply:
Thanks for clarify! Your approach looks good. Feel free to close the PR and push new one.